File: python/with_java/programmers/level 2/할인 행사.py
def check(info):
    for key in info:
        if info[key] != 0:
            return False
    return True

def solution(want, number, discount):
    info = {}
    for i in range(len(want)):
        info[want[i]] = number[i]
    for i in range(9):
        if discount[i] in info:
            info[discount[i]] -= 1
    answer = 0
    for i in range(9, len(discount)):
        if discount[i] in info:
            info[discount[i]] -= 1
        if check(info):
            answer += 1
        if discount[i-9] in info:
            info[discount[i-9]] += 1
    return answer

File: java/src/main/java/ps_traning/programmers/level_2/할인_행사.java
package ps_traning.programmers.level_2;

import java.util.Arrays;
import java.util.Collections;

public class 할인_행사 {
    public int solution(String[] want, int[] number, String[] discount) {
        int answer = 0;
        for (int i = 10; i <= discount.length; i++) {
            String[] temps = Arrays.copyOfRange(discount, i - 10, i);
            boolean flag = true;
            for (int j = 0; j < want.length; j++) {
                String item = want[j];
                if (Collections.frequency(Arrays.asList(temps), item) != number[j]) {
                    flag = false;
                    break;
                }
            }
            if (flag) {
                answer++;
            }
        }
        return answer;
    }
}
